### PR TITLE
added sort-pipe

### PR DESCRIPTION
**Pull request by Catalin12**

### CODE REVIEW

File: component-interaction/src/app/sort-users.pipe.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import { Pipe, PipeTransform } from "@angular/core";
+
+import { UserModel } from './user.model';
+
+@Pipe({ name: "sortUsers" })
+export class SortUsers implements PipeTransform {
```
Comment (danielursu) [inline component-interaction/src/app/sort-users.pipe.ts]:
```suggestion
export class SortUserPipe implements PipeTransform {
```

File: component-interaction/src/app/sort-users.pipe.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+	public transform(value: UserModel[], args?: string) {
```
Comment (danielursu) [inline component-interaction/src/app/sort-users.pipe.ts]:
```suggestion
	public transform(value: UserModel[], orderType: OrderTypeEnum) {
```

File: component-interaction/src/app/user-list/user-list.component.ts [hunk 3]
```diff
@@ -65,6 +70,7 @@ export class UserListComponent implements OnInit {
 		};
 		this.users.push(user);
 		this.id++;
+		this.users = [...this.users]; //we do this to sort again list with users
```
Comment (danielursu) [inline component-interaction/src/app/user-list/user-list.component.ts]: also check for deletion of a user

Comment (danielursu) [inline component-interaction/src/app/user-list/user-list.component.ts]:
```suggestion
		this.users = [...this.users]; //we do this to change the reference of the array in order to enforce the CD (change detection)
```

File: component-interaction/src/app/sort-users.pipe.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+		console.log(`sortUsers was called.`);
+		if (args === "DSC") {
```
Comment (danielursu) [inline component-interaction/src/app/sort-users.pipe.ts]:
```suggestion
		if (orderType === OrderTypeEnum.DSC) {
```

File: component-interaction/src/app/app.module.ts [hunk 5]
```diff
@@ -2,14 +2,16 @@ import { NgModule } from '@angular/core';
 import { BrowserModule } from '@angular/platform-browser';
 
 import { AppComponent } from './app.component';
+import { SortUsers } from './sort-users.pipe';
 import { UserDetailsComponent } from "./user-details/user-details.component"
 import { UserListComponent } from './user-list/user-list.component';
```
Comment (danielursu) [inline component-interaction/src/app/app.module.ts]: change to double quotes

**Review by danielursu — CHANGES_REQUESTED** [review]

File: component-interaction/src/app/user-list/user-list.component.ts (hunk 3 above)
Comment (Catalin12) [reply component-interaction/src/app/user-list/user-list.component.ts]: I don t think its necessary

File: component-interaction/src/app/app.module.ts [hunk 6]
```diff
@@ -1,20 +1,22 @@
 import { NgModule } from '@angular/core';
 import { BrowserModule } from '@angular/platform-browser';
 
-import { AppComponent } from './app.component';
+import { AppComponent } from "./app.component";
```
Comment (danielursu) [inline component-interaction/src/app/app.module.ts:4]: when you have to replace in many places you should use CTRL+F -> Replace all

**Review by danielursu — APPROVED** [review]

**Pull request merged**